### PR TITLE
Replace Prismjs and manual code editing system with CodeMirror

### LINKED ISSUE #774: Night mode contrast of blue text not readable.
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

Night mode contrast of blue text not readable. The rotate kw for example.

### What did you expect to see?

Expected to read it properly.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/transform-function/rotate`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/transform-function/rotate/index.md
* Last commit: https://github.com/mdn/content/commit/345cf4c7023bae3a64904f6a27fc5171a6d2a6ae
* Document last modified: 2022-04-19T21:12:12.000Z

</details>

### LINKED ISSUE #811: Color of css ID selector
ID selectors in tabbed examples are unreadable in dark mode because of poor color choice. Right now color #30a is used, which makes selector look like that:

image

I don't think there is any example using id selector right now, but I am about to add one in ::part pseudo-element.

### LINKED ISSUE #813: poor code color contrast with css function names in dark mode
The CSS code in `.example-choice` boxes sometimes have poor color contrast in dark mode.
This can be an issue with:
- function names
- the values inside function parenthesis

just a few examples:

https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path

`clip-path` function text and the path values themself have poor contrast

image

-----------------

I assume this is an issue with all gradient function names

https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/radial-gradient

poor-contrast

### LINKED ISSUE #858: Dark mode: Some keywords don't have enough contrast in some interactive examples
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/calc

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/CSS/calc#try_it

### What information was incorrect, unhelpful, or incomplete?

In dark mode it's hard to see the `calc` and `var` keywords.

image

### What did you expect to see?

I expect to see those keywords with more contrast.

### Do you have any supporting links, references, or citations?

image

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/calc`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/calc
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/calc/index.md
* Last commit: https://github.com/mdn/content/commit/f01246f406289c4c5222a5f340bcb8668118592a
* Document last modified: 2022-04-06T08:08:41.000Z

</details>

### PR DESCRIPTION
**Pull request by queengooborg**
This PR replaces the Prism code highlighting and manual code editing system with CodeMirror.  This significantly reduces the dependency count and simplifies our code.  Closes #886 due to irrelevance.  Naturally fixes #858, fixes #774, fixes #813, fixes #811 by using the CodeMirror theme.

This will currently block work for #851.

### CODE REVIEW

**Comment by caugner** [comment]
@queengooborg As this PR changes how the interactive examples look, could you please add Before/After screenshots for each of the 4 editor types? 🙏

**Comment by queengooborg** [comment]
Screenshots now attached!  This actually _only_ affects the CSS examples (all of the other editors use CodeMirror)!

Before:
Screen Shot 2022-09-22 at 04 16 22
(Note that the black text on dark theme is due to a CSS issue that will be resolved in another PR if this one doesn't land.)

After:
Screen Shot 2022-09-22 at 04 14 42

**Comment by caugner** [comment]
I think the reason for using Prismjs here was to make the interactive examples look like the code samples (which will continue to use Prism): https://developer.mozilla.org/en-US/docs/Web/CSS/animation#applying_multiple_animations

Not really sure how to solve this. 🤔

**Comment by queengooborg** [comment]
Sadly, from what I noticed, it seemed as though Prismjs never actually ran on the interactive examples -- accessing a random CSS page with an interactive example, the text is a uniform color: https://developer.mozilla.org/en-US/docs/Web/CSS/transition

Screen Shot 2022-09-23 at 02 48 03

Additionally, once I got it working again, I ran into some other unfortunate issues that deteriorate the user experience, such as: if we perform syntax highlighting while typing, the user loses their undo state and it's quite hard to retain the text cursor position when the HTML changes, but if we don't, the user will end up getting colors bleeding over into parts it shouldn't (if that makes any sense).

Personally, I think that the benefits to using CodeMirror everywhere outweigh the drawbacks, especially when considering that this same issue applies to all of the existing CodeMirror editors.

**Comment by queengooborg** [comment]
@schalkneethling @caugner May I get a review on this soon?

File: editor/js/editor-libs/events.js [hunk 1]
```diff
@@ -8,14 +8,14 @@ import * as cssEditorUtils from "./css-editor-utils.js";
...
-  exampleChoiceList.addEventListener("paste", handlePasteEvents);
 
   exampleChoiceList.addEventListener("keyup", function (event) {
     var exampleChoiceParent = event.target.parentElement;
 
     cssEditorUtils.applyCode(
       exampleChoiceParent.textContent,
-      exampleChoiceParent
+      // keyup events fire on the .cm-scroller four levels down from the choice div
+      exampleChoiceParent.parentElement.parentElement.parentElement
```
Comment (caugner) [inline editor/js/editor-libs/events.js]: Any chance we can use [`closest()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) to get the desired element?

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM, just one possible nit.

File: editor/js/editor-libs/events.js (hunk 1 above)
Comment (queengooborg) [reply editor/js/editor-libs/events.js]: I had a feeling something like this existed but wasn't sure what it was called, thanks!

**Review by caugner — APPROVED** [review]

**Pull request merged**